Source/DafnyCore/Backends/ExecutableBackend.cs

```diff
@@ -50,26 +50,49 @@ public abstract class ExecutableBackend : IExecutableBackend {
   }
 
   protected void ProcessOuterModules(Program dafnyProgram) {
-    var outerModules = GetOuterModules();
-    ModuleDefinition rootUserModule = null;
-    foreach (var outerModule in outerModules) {
-      var newRoot = new ModuleDefinition(RangeToken.NoToken, new Name(outerModule), new List<IToken>(),
-        ModuleKindEnum.Concrete, false,
-        null, null, null);
-      newRoot.EnclosingModule = rootUserModule;
-      rootUserModule = newRoot;
+    // Apply the --outer-module option from any translation records for libraries,
+    // but only to top-level modules.
+    // TODO: Check the invariant that a module isn't in both a translation record
```

Comment:
```
If (hasTranslationRecord) {
  // use it
} else {
 var outerModuleName = Options.Get(OuterModule);
  ..
}
```

Reply:
Even better this is checked in Validate and Merge now instead.